tsconfig.eslint.json

```diff
@@ -0,0 +1,12 @@
+{
```

Comment:
This file can also go...

Reply:
This is the tsconfig file for [TypeScript ESLint](https://github.com/typescript-eslint), the reason why there are two of these files is because we need a way for TypeScript ESLint to know which files it is supposed to lint while not imposing on the TypeScript compiler's config file &mdash it extends the base for linting purposes.

Believe it or not, this is actually the way that TypeScript ESLint recommends doing it and they even do it themselves in [their own repository](https://github.com/typescript-eslint/typescript-eslint).

Reply:
That is for the Typescript eslint extension.

Reply:
Yes, the TypeScript ESLint extension can lint JS code. The reason why this works out is that TS is _supposed_ to be a gradually typable superset of JS, so many of the best practices that it enforces are applicable to JS. I wish we could've gone over this PR yesterday, it's quite a fascinating topic.

Reply:
It's not actually one tho; it seems very strange to be using the TS parser on "not TS".

Reply:
What were you thinking for the parser then? I used to use `babel-eslint`, but then you can't make use of TypeScript ESLint and you need to configure Babel too, which is no fun.

Reply:
> TS is supposed to be a gradually typable superset of JS

Sorry, to clarify what I meant with this is that the places where it diverges (array index signatures and symbols behavior) need to be avoided to make proper use of this parser in pure JS.

Reply:
It’s regular JavaScript - why not the core parser? If we need stage 3 features, babel-eslint (which can be configured with preset-env and nothing else) should be fine. Configuring babel is easier than configuring TS :-)

Reply:
I hope to have clarified the above in https://github.com/tc39/tc39.github.io/pull/213#discussion_r505023589.

Reply:
The recommended plugin for verifying JSDoc is https://github.com/gajus/eslint-plugin-jsdoc

Please see the end of life documentation for eslint's builtin JSDoc support: https://eslint.org/blog/2018/11/jsdoc-end-of-life#suggested-replacement

Reply:
Conclusion: we can't use Esprima as the parser because it doesn't seem to support top-level await yet.

```console
/mnt/c/orgs/tc39/tc39.github.io/_tasks/check-html.mjs
  4:20  error  Parsing error: Cannot use keyword 'await' outside an async function
```

This is will have to use `babel-eslint` then.